src/web/mjs/connectors/MangaJar.mjs

```diff
@@ -21,40 +21,48 @@ export default class MangaJar extends Connector {
 
     async _getMangas() {
         let mangaList = [];
-        for(let page = 1, run = true; run; page++) {
+        for (let page = 1, run = true; run; page++) {
             let mangas = await this._getMangasFromPage(page);
             mangas.length > 0 ? mangaList.push(...mangas) : run = false;
         }
         return mangaList;
     }
 
-    async _getMangasFromPage( page ) {
+    async _getMangasFromPage(page) {
         let request = new Request(new URL(`/manga?page=${page}`, this.url));
         let data = await this.fetchDOM(request, 'div.row article.flex-item div.post-description a.card-about');
         return data.map(element => {
             return {
                 id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.querySelector( 'p.card-title' ).textContent.trim()
+                title: element.querySelector('p.card-title').textContent.trim()
             };
         });
     }
 
-    async _getChapters( manga ) {
+    async _getChapters(manga) {
         let request = new Request(new URL(manga.id.replace(/\/$/, '') + '/chaptersList', this.url));
-        let data = await this.fetchDOM(request, 'ul.list-group li.chapter-item a');
-        return data.map(element => {
-            return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.innerText.trim(),
-                language: ''
-            };
-        });
+        let pages = await this.fetchDOM(request, 'div.chapters-infinite-pagination ul.pagination a:last-of-type');
+        let pageCount = pages.length > 0 ? parseInt(pages[pages.length - 2].href.match(/p=(\d+)$/)[1]) : 1;
+        let chapterList = [];
+        for (let page of new Array(pageCount).keys()) {
+            let request = new Request(new URL(this.url + manga.id.replace(/\/$/, '') + '/chaptersList' + '?page=' + (page + 1)), this.requestOptions);
```

Comment:
May declare the url outside of the loop and inside the loop just set the page, e.g.
```javascript
const uri = new URL(manga.id + '/chapterlist', this.url)
for let page ... {
    uri.searchParams.set('page', page + 1)
    const request = new Request(uri, ...)
```

```